### ISSUE TITLE
Server initialization failed.

### ISSUE THREAD

**Issue by CoreySchnedl** [cucumber/vscode#234]
### 👓 What did you see?

[Error - 12:45:07 PM] Server initialization failed.
  Message: Request initialize failed with message: bad export type for `tree_sitter_c_sharp_external_scanner_create`: undefined
  Code: -32603 
[Error - 12:45:07 PM] Cucumber Language Server client: couldn't create connection to server.
  Message: Request initialize failed with message: bad export type for `tree_sitter_c_sharp_external_scanner_create`: undefined
  Code: -32603 

Using cucumber v1.10.0
Using vscode in WSL v1.90.0

### ✅ What did you expect to see?

The extension working. It is currently not usable. 

### 📦 Which tool/library version are you using?

Using cucumber v1.10.0
Using vscode in WSL v1.90.0

### 🔬 How could we reproduce it?

1. Install extension
2. Restart vscode
3. See error [Error - 12:45:07 PM] Server initialization failed.

**Comment by dcuenca-fluidra** [comment]
Same here. We had to downgrade VSCode to 1.89 and disable auto updates to continue having this great plugin working.

**Comment by jclery-pass** [comment]
Same problem here.

**Comment by barrymichaeldoyle** [comment]
This has been a problem for a while now 😞 I saw VSCode just released a new update and it is still not working.

**Comment by elDin-Dev** [comment]
Yes, i test every 2 days with VS Code insider (1.92.0-insider) the problem still there

**Comment by jhassine** [comment]
This also happens inside a devcontainer

**Comment by martin-shields-sage** [comment]
I have the same issue as well

**Comment by jm-paulin** [comment]
Would cucumber/language-service#209 help to address this issue @kieran-ryan ?

**Comment by rolger** [comment]
It also fails for me with the same message. Maybe this could help:

I setup an enviroment in gitpod and the same error occured. Then I tried the same on my Window 10 environment and it worked fine. So, really was wondering what is the difference. Then I used a local installation of Ubuntu and it fails again. So it seems that it depends on the operating system.

```
Version: 1.92.1 (system setup)
Commit: eaa41d57266683296de7d118f574d0c2652e1fc4
Date: 2024-08-07T20:16:39.455Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631
```

and 

```
Extensions installed on WSL: Ubuntu:
cucumberopen.cucumber-official@1.10.0
formulahendry.dotnet-test-explorer@0.7.8
ms-azuretools.vscode-docker@1.29.2
ms-dotnettools.csharp@2.39.29
ms-dotnettools.vscode-dotnet-runtime@2.1.1
redhat.java@1.33.0
visualstudioexptteam.intellicode-api-usage-examples@0.2.8
visualstudioexptteam.vscodeintellicode@1.3.1
vscjava.vscode-gradle@3.16.4
vscjava.vscode-java-debug@0.58.0
vscjava.vscode-java-dependency@0.24.0
vscjava.vscode-java-pack@0.29.0
vscjava.vscode-java-test@0.42.0
vscjava.vscode-maven@0.44.0
```

**Comment by guenhter** [comment]
I also get this problem. What is the right location to address this issue: Is it the cucumber-language-server, it is tree-sitter? I guess it is not the vscode cucumber plugin itself, is it?

**Comment by Ramin2c** [comment]
I just tested and the issue is still there in **1.90.0, 1.91.0 and 1.92.0**
Downgraded to **1.89.0** and it works like it should.

Used winget to install an older version:

`winget install -e --id Microsoft.VisualStudioCode -v 1.89.0`

**_NOTE:_** the current version needs to uninstalled, either manually or through Programs

**Comment by jm-paulin** [comment]
Just had the **vscode v1.93.0** update, and same issue.... 😭 😭 😭

**Comment by ViniciusResende** [comment]
Facing the same issue with the following:
**WSL**
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.22631.4169

**Visual Studio**
Version: 1.93.1 (user setup)
Commit: 38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40
Date: 2024-09-11T17:20:05.685Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631

**Linux**
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy

**Comment by naadev** [comment]
Same here:

Version: 1.94.0 (user setup)
Commit: d78a74bcdfad14d5d3b1b782f87255d802b57511
Date: 2024-10-02T13:08:12.626Z
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045

Cucumber VSIX
v1.10.0

**Comment by romainwg** [comment]
Same here:

WSL:

WSL version: 2.3.26.0
Kernel version: 5.15.167.4-1
WSLg version: 1.0.65
MSRDC version: 1.2.5620
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.26100.2314
Node version: v21.6.2

Visual Studio Code:

1.95.3
f1a4fb101478ce6ec82fe9627c43efbf9e98c813
x64

Installed Extensions:

cucumberopen.cucumber-official@1.10.0

Linux:

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy

**Comment by mhvelplund** [comment]
Still broken in:
```
Version: 1.95.3
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.19045
```

... on a related note: it's bad enough I have to write unit tests in the 🫏backwards format, but when the plugin that makes it bearable stops working ... I'm not ashamed to say I had a little cry :)

**Comment by TriPawdCat** [comment]
Please could we have an update on this Bug?  This has been open for 6 months and appears that no progress has been made.  
VSCode is the IDE of choice for our organisation and working without this extension is now becoming very difficult and we are now considering migrating away from using Cucumber for our BDD Test Framework and will need to look alternatives.

I hope that there is an update in the very near future :-)

**Comment by guenhter** [comment]
I wonder if this plugin works for anyone.

We use `alexkrechik.cucumberautocomplete` as an alternative.

**Comment by Ramin2c** [comment]
> I wonder if this plugin works for anyone.
> 
> We use `alexkrechik.cucumberautocomplete` as an alternative.

How does it work on your side? For example:

1. Does it help you generate the test skeleton?
2. Does it have autocomplete in scenario files?
3. Does it link scenarios to the python code?

**Comment by guenhter** [comment]
To be honest, I don't use this features much. My only use case is the syntax highlighting and auto formatting.

**Comment by Ramin2c** [comment]
"syntax highlighting and auto formatting" is equally important, I'd like to
give it a try as well. Thanks for suggesting the extension.

**Comment by Ramin2c** [comment]
> To be honest, I don't use this features much. My only use case is the syntax highlighting and auto formatting.

"syntax highlighting and auto formatting" is equally important, I'd like to
give it a try as well. Thanks for suggesting the extension.

**Comment by bmeier-pros** [comment]
Yes, @guenhter thank you for the suggestion.  That plugin works for our use cases currently, though it has some quirks.  The official cucumber plugin is apparently not going to be maintained further, so we've switched to https://marketplace.visualstudio.com/items?itemName=alexkrechik.cucumberautocomplete for our BDD work.

**Comment by exetroncorp** [comment]
is this project  dead ?

**Comment by mfreibergs** [comment]
> is this project dead ?

Sadly looks like it. New maintainers are desperately needed!

**Comment by luke-hill** [comment]
A new build of the extension should be coming out soon. I just need to ensure the build process for releasing this works the same as the other polyglot services @romaindup - Sorry for the delay.

**Comment by jm-paulin** [comment]
Thank you @luke-hill !!!
